tokio-threadpool/src/worker/mod.rs

```diff
@@ -79,8 +79,11 @@ struct CurrentTask {
 ///
 /// This identifier is unique scoped by the thread pool. It is possible that
 /// different thread pool instances share worker identifier values.
-#[derive(Debug, Clone, Hash, Eq, PartialEq)]
-pub struct WorkerId(pub(crate) usize);
+///
+/// Worker identifiers in a single thread pool are guaranteed to be integers in
+/// the range `0..pool_size`.
+#[derive(Debug, Clone, Copy, Hash, Eq, PartialEq)]
+pub struct WorkerId(pub usize);
```

Comment:
Instead of making the field `pub`, could you instead keep the field private and add a new fn: `WorkerId::as_usize()`? This should help with forwards compatibility.

Reply:
I decided to name it `WorkerId::into_usize` because:

* `fn as_usize(&self) -> &usize`
* `fn to_usize(&self) -> usize`
* `fn into_usize(self) -> usize`

I made `WorkerId` a `Copy` type (just like e.g. [`ThreadId`](https://doc.rust-lang.org/nightly/std/thread/struct.ThreadId.html) is) so now `into_usize` seems most fitting.